src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/SourceGeneratorAttributeAnalyzerFix.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = nameof(SourceGeneratorAttributeAnalyzerFix))]
+    [Shared]
+    public sealed class SourceGeneratorAttributeAnalyzerFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            DiagnosticIds.MissingSourceGeneratorAttributeId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
```

Comment:
I'm not sure if this comment (https://github.com/dotnet/roslyn-analyzers/issues/192#issuecomment-136806431) about `CodeAction.Create` still applies.

Reply:
hmm, if that's the case there's at least one other place that needs to be fixed. Let me look into how telemetry id gets calculated

Reply:
So... yes and no (ish) 

https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs#L192 

I think that's the function that's getting telemetry for suggested code fixes. It has `TryGetTelemetryId(out var telemetryId)` which does have the problem described in that comment. It only uses the type as an indicator for telemetry. 

If the diagnostic id is also being saved (https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs#L208) then we have enough information to distinguish this code fix. The downside would be if we added more code fixes we might lose which one is being applied. 

In short, I think adding a class is best... even if I don't like it :(

Reply:
Use `CodeAction.Create` to match the rest of this package.